homework-2/services/userService.ts

```diff
@@ -1,83 +1,57 @@
-import { IUser } from "../interfaces/IUser";
-import { v4 as uuidv4 } from 'uuid';
-
-const initialUsersList: IUser[] = [
-    {
-        id: '14f4e1a1-ff87-4f93-b01f-2293f82f1345',
-        login: 'user1query1',
-        password: 'user1',
-        age: 1,
-        isDeleted: false
-    },
-    {
-        id: 'cbe11495-1dc1-489b-9a1b-5d215e6501a9',
-        login: 'user2query12123123123',
-        password: 'user2',
-        age: 2,
-        isDeleted: false
-    },
-    {
-        id: 'fe1ed5a7-7c47-41fe-9248-f776bd706b18',
-        login: 'user3query123',
-        password: 'user3',
-        age: 3,
-        isDeleted: false
-    },
-    {
-        id: '4b4ba4ab-0f8c-4669-9aef-0bcf220dad8e',
-        login: 'user4',
-        password: 'user4',
-        age: 4,
-        isDeleted: false
-    }
-];
+import { IUserAttrs } from "../interfaces/IUserAttrs";
+import { IUserModel } from "../models/userModel";
+import { db } from "../models";
 
-class UserService {
-    private usersList: IUser[];
 
-    constructor() {
-        this.usersList = initialUsersList;
-    }
-
-    getAllUsers(): IUser[] {
-        return this.usersList;
+class UserService {
+    getAllUsers(): Promise<IUserAttrs[]> {
+        return db.User.findAll()
     }
 
-    getUserById(id: string): IUser | null {
-        return this.usersList.find(user => user.id === id) || null;
+    getUserById(id: string): Promise<IUserModel | null> {
+        return db.User.findByPk(id);
     }
 
-    addUser(user: IUser): string {
-        user.id = user.id || uuidv4();
-        this.usersList.push(user);
+    async addUser(user: IUserAttrs): Promise<number> {
```

Comment:
I would add method getUserByLogin and move checking part into controller level

Reply:
Done